src/darwin/CHIPTool/CHIPTool/Framework Helpers/DefaultsUtils.m

```diff
@@ -115,23 +116,50 @@ BOOL CHIPGetConnectedDeviceWithID(uint64_t deviceId, CHIPDeviceConnectionCallbac
 
 BOOL CHIPIsDevicePaired(uint64_t deviceId)
 {
-    CHIPDeviceController * controller = InitializeCHIP();
+    NSString * PairedString = CHIPGetDomainValueForKey(kCHIPToolDefaultsDomain, KeyForPairedDevice(deviceId));
+    return [PairedString boolValue];
+}
 
-    NSError * error;
-    bool paired = [controller isDevicePaired:deviceId error:&error];
-    if (error != nil) {
-        NSLog(@"Error retrieving device info for deviceId %llu", deviceId);
-        paired = NO;
-    }
-    return paired;
+void CHIPSetDevicePaired(uint64_t deviceId, BOOL paired)
+{
+    CHIPSetDomainValueForKey(kCHIPToolDefaultsDomain, KeyForPairedDevice(deviceId), paired ? @"YES" : @"NO");
 }
 
+NSString * KeyForPairedDevice(uint64_t deviceId) { return [NSString stringWithFormat:@"%@%llu", kDevicePairedKey, deviceId]; }
+
 void CHIPUnpairDeviceWithID(uint64_t deviceId)
 {
-    CHIPDeviceController * controller = InitializeCHIP();
-
-    NSError * error;
-    [controller unpairDevice:deviceId error:&error];
+    CHIPSetDevicePaired(deviceId, NO);
+    CHIPGetConnectedDeviceWithID(deviceId, ^(CHIPDevice * _Nullable device, NSError * _Nullable error) {
+        if (error) {
+            NSLog(@"Failed to unpair device %llu still removing from CHIPTool. %@", deviceId, error);
```

Comment:
Per conversation with @sagar-apple this is a purposeful decision for this UI flow.

Reply:
So I felt that this was an application decision here. This is what I want CHIPTool to do when the user requests to remove a pairing. 

The application will just "forget" the pairing and attempt to inform the server of it as well. In a case where the server is reset without the controller's command (manual reset) it will just leave a bunch of stale data inside CHIPTool. 

So to keep things clean, CHIPTool will immediately unpair but attempt to notify the server of the removal as well.